opm/simulators/flow/BlackoilModelNldd.hpp

```diff
@@ -447,7 +456,9 @@ class BlackoilModelNldd {
             // Check for local convergence.
             detailTimer.reset();
             detailTimer.start();
+            resnorms.clear();
             convreport = this->getDomainConvergence(domain, timer, iter, logger, resnorms);
+            convergence_history.push_back(resnorms);
```

Comment:
You might potentially write `getDomainConvergence(/* ... *, convergence_history.emplace_back())` here.  That would obviate the `clear() + push_back()` operations.  Writing this requires C++17 and later, of course, because the signature of `vector<>::emplace_back()` changed to return a `vector<>::reference` to the new element in C++17.

Reply:
I had to think about this for some time before I understood exactly what would happen. I think you are right, this would work, but for now I prefer the original as it is simpler for the reader. If the "pass the reference returned by a default-constructed just-emplaced element" pattern becomes widespread I may change my mind, but for now I consider it a little exotic.